airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -620,10 +629,10 @@ def create_legacy_to_per_partition_state_migration(
             )
 
         return LegacyToPerPartitionStateMigration(
-            declarative_stream.retriever.partition_router,
-            declarative_stream.incremental_sync,
+            partition_router,
+            declarative_stream.incremental_sync,  # type: ignore # was already checked. Migration can be applied only to incremental streams.
             config,
-            declarative_stream.parameters,
+            declarative_stream.parameters,  # type: ignore
```

Comment:
Why do we need to ignore here?

Reply:
different types are expected there `Mapping[str, Any] != Dict[str, Any]`

Reply:
Yeah, the difference in typing is annoying... I'm fine with this "ignore" then. Can we add a comment to explain this?

Reply:
added